homeassistant/components/matter/adapter.py

```diff
@@ -30,6 +101,28 @@ def get_clean_name(name: str | None) -> str | None:
     return name.strip() or None
 
 
+def _get_connections(endpoint: MatterEndpoint) -> set[tuple[str, str]]:
+    """Get connections for device registry."""
+    network_interfaces: list[GeneralDiagnostics.Structs.NetworkInterface] = (
+        endpoint.get_attribute_value(
+            None, GeneralDiagnostics.Attributes.NetworkInterfaces
+        )
+        or []
+    )
+
+    connections: set[tuple[str, str]] = set()
+    for ni in network_interfaces:
+        if hardwareAddress := ni.hardwareAddress:
+            try:
+                if connection := HardwareAddress.from_bytes(
+                    address=hardwareAddress
+                ).connection():
+                    connections.add(connection)
+            except ValueError:
+                continue
```

Comment:
This was all I meant with my review feedback - just a try..except around the code that reads the mac address into a mac address string (as that can be an empty string or malformed junk) to not blow up setting up a device

Reply:
`GeneralDiagnostics.Structs.NetworkInterface.hardwareAddress` are (possibly empty) `bytes`. I guard against `None` and empty here.

`convert_mac_address` takes these bytes, formats and concatenates them. The result is some string in the format (`"xx:xx:xx:..."`).

I check that the string has an appropriate length for EUI-48 or EUI-64 addresses and set the connection type accordingly. Further normalisation is done by the HA core code.

As far as I can see, there is nothing that can go wrong, except if the matter/chip library itself is buggy and doesn't adhere to their types. 

I don't know what to catch except a broad `Exception` and move on.

Reply:
Let's do it this way, you guard for empty values which is (imo) enough. If something unexpected can happen later, we will learn soon enough and can act on it then.

BTW: I still like your proposal for the validation of the addresses so I advice you to try raising that discussion and/or open a draft PR to fix the device registry.